kolibri/core/assets/src/state/actions.js

```diff
@@ -223,6 +223,36 @@ function getCurrentSession(store) {
   .catch(error => { handleApiError(store, error); });
 }
 
+function getFacilityConfig(store) {
+  const coreApp = require('kolibri');
+  const FacilityCollection = coreApp.resources.FacilityResource
+    .getCollection()
+    .fetch();
+
+  return FacilityCollection.then(facilities => {
+    const currentFacilityId = facilities[0].id;
+    const facilityConfigCollection = coreApp.resources.FacilityDatasetResource
+      .getCollection({ facility_id: currentFacilityId })
+      .fetch();
+    return facilityConfigCollection
```

Comment:
It seems like most screens are basically following the Django workflow (Model -> HTML string), but with a extra steps (Model -> Serialized Model -> Deserialized Model -> Slightly Transformed (i.e. camelCased) Model -> Entry in Vuex -> Vue View).

By having this many boundaries, there's going to be some ambiguity about where certain data transformations should (and could) take place (at the model/viewset/serializer level, inside the vuex action, inside the view?).

I'm sort of used to it now, but because of all this indirection, it sometimes takes a bit of detective work to understand where the data on the UIs come from. By actually dumping the API data as-is into the view or with a minimal identity-function/camelCaser transformation, it actually removes one layer of indirection, because I can go look the REST API docs, or the network tab and understand the data.

If the API data is assumed to be immutable, there are almost no pitfalls to this, except maybe overfetching data, which could be addressed by providing a hook on the Resource class to select which fields one wants.